library/src/main/java/com/google/android/exoplayer/hls/HlsChunkSource.java

```diff
@@ -573,7 +624,16 @@ private void setMediaPlaylist(int variantIndex, HlsMediaPlaylist mediaPlaylist)
     variantLastPlaylistLoadTimesMs[variantIndex] = SystemClock.elapsedRealtime();
     variantPlaylists[variantIndex] = mediaPlaylist;
     live |= mediaPlaylist.live;
-    durationUs = live ? C.UNKNOWN_TIME_US : mediaPlaylist.durationUs;
+    liveEvent |= mediaPlaylist.liveEvent;
+    durationUs = live || liveEvent ? C.UNKNOWN_TIME_US : mediaPlaylist.durationUs;
+
+    if(!live || liveEvent) {
```

Comment:
@ojw28 @thasso When I remove the conditional at line 630 and then test against the iTV24 live stream, it sort of works correctly.  There is an excessive delay in forward seeking but backward seeking works fine.  Can either of you make a suggestion where I can find and attempt to fix this seek bug.  I have a hunch it is a known, existing issue.